docs/investigations/secure-resolvers/secure-auth-resolvers.md

```diff
@@ -0,0 +1,184 @@
+# Limit access to GraphQL resources
+
+This document describes proposed solution to the problem of unlimited resource access.
+Limiting access to certain resolvers is important especially in case of accessing authorization details.
+
+## Terminology
+
+API Consumer - Application / Runtime / Integration System / User 
+
+## Problems we want to solve
+
+### Restrict access to `application { api { auth(runtimeID) } }` for Runtimes with ID different than passed `runtimeID`
+
+RuntimeAgent wants to read auth details for application's API.
+The RuntimeAgent should be allowed to read field `auth` only with his own ID.
+In this case we will use new directive `limitAccessFor` and `limitIntegrationSystemAccessTo`.
+
+### Restrict access to `application { auths }`
+
+RuntimeAgent wants to read applications but shouldn't have access to their auth details.
+We can restrict the access by current `hasScopes` directive.
+
+### Restrict access to resolvers for specific types of API consumers (by ID)
+
+We should be able to restrict access to any resolver that takes API Consumer ID as one of its parameters by adding directive.
+
+We want to cover following cases:
+
+1. Runtime musn't read other Runtimes.
+2. Application musn't read other Applications.
+3. Integration System musn't read other Integration Systems.
+4. Runtime musn't read Applications which are not in the same scenario (`applicationsForRuntime`).
+
+### Restrict access to Application/Runtime for IntegrationSystem is which not managed
+When Integration System request specific Application/Runtime, it should be able to read only object managed by itself.
+
+## Solution
+To achieve those restrictions the following solution is proposed:
+
+### Add information about API consumer in Tenant Mapping service
+Tenant Mapping Service adds 2 items to the JWT token
```

Comment:
This is already done, so maybe you can rephrase this that it will not be added but basically just make use of those two items.

Reply:
```suggestion
We can use information which is added in Tenant Mapping Service to JWT [PR](https://github.com/kyma-incubator/compass/pull/475):
```